includes/Core/Site_Health/Tag_Placement.php

```diff
@@ -221,7 +220,12 @@ function( $module ) {
 	 */
 	protected function check_if_tag_exists( $module, $content ) {
 		$check_tag   = $module->has_placed_tag_in_content( $content );
-		$module_name = $module->get_public_name();
+		$module_name = $module->name;
+
+		// Remove 4 from Analytics name if we are checking Analytics 4 module.
+		if ( strpos( $module_name, '4' ) !== false ) {
```

Comment:
Let's check if it is the GA4 module specifically rather than doing this kind of a loose match unnecessarily.

```suggestion
		if ( $module instanceof Analytics_4 ) {
```